ram_report fails due to pyelftools version

**Describe the bug**
Not able to get ram_report when using pyelftools=0.27.

**To Reproduce**
Steps to reproduce the behavior:
1. rm -rf build; west build -b esp32c3_devkitm samples/hello_world
2. west build -t ram_report
3. Error:
```c
~/espressif/zephyr/zephyr (declare-off-t-and-ssize-t-in-fdtable-h) $ west build -t ram_report
-- west build: running target ram_report
Traceback (most recent call last):
  File "/home/sylvio/espressif/zephyr/zephyr/scripts/footprint/size_report", line 876, in <module>
    main()
  File "/home/sylvio/espressif/zephyr/zephyr/scripts/footprint/size_report", line 852, in main
    do_simple_name_matching(dwarfinfo, symbol_dict, processed)
  File "/home/sylvio/espressif/zephyr/zephyr/scripts/footprint/size_report", line 361, in do_simple_name_matching
    symbol = match_symbol_address(symlist, die,
  File "/home/sylvio/espressif/zephyr/zephyr/scripts/footprint/size_report", line 115, in match_symbol_address
    low, high = get_die_mapped_address(die, parser, dwarfinfo)
  File "/home/sylvio/espressif/zephyr/zephyr/scripts/footprint/size_report", line 85, in get_die_mapped_address
    loc = parser.parse_from_attribute(loc_attr, die.cu['version'], die)
TypeError: LocationParser.parse_from_attribute() takes 3 positional arguments but 4 were given
FAILED: zephyr/cmake/reports/CMakeFiles/ram_report /home/sylvio/espressif/zephyr/zephyr/build/zephyr/cmake/reports/CMakeFiles/ram_report 
cd /home/sylvio/espressif/zephyr/zephyr/build && /usr/bin/python3 /home/sylvio/espressif/zephyr/zephyr/scripts/footprint/size_report -k /home/sylvio/espressif/zephyr/zephyr/build/zephyr/zephyr.elf -z /home/sylvio/espressif/zephyr/zephyr -o /home/sylvio/espressif/zephyr/zephyr/build --workspace=/home/sylvio/espressif/zephyr -d 99 ram
ninja: build stopped: subcommand failed.
```

**Expected behavior**
ram_report output working

**Environment (please complete the following information):**
 - OS: Linux SDK 0.16.8
 - v3.7.0-rc2

--- a/scripts/requirements-base.txt
+++ b/scripts/requirements-base.txt
@@ -4,7 +4,7 @@
 # part of the recommended workflow
 
 # used by various build scripts
-pyelftools>=0.27
+pyelftools>=0.29
 
 # used by dts generation to parse binding YAMLs, also used by
 # twister to parse YAMLs, by west, zephyr_module,...

